Incompatibility of python ismrmrd with xsdata causes test failures

```
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/data/_IsmrmrdRawTestData.py:213: in __init__
    dset.write_xml_header(header.toXML('utf-8'))
/opt/hostedtoolcache/Python/3.11.8/x64/lib/python3.11/site-packages/ismrmrd/xsd/pyxb_compat.py:18: in ToXML
    config = SerializerConfig(encoding=encoding,pretty_print=True)
<string>:11: in __init__
    ???
/opt/hostedtoolcache/Python/3.11.8/x64/lib/python3.11/site-packages/xsdata/formats/dataclass/serializers/config.py:40: in __post_init__
    self.__setattr__("pretty_print", pretty_print)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = SerializerConfig(encoding='utf-8', xml_version='1.0', xml_declaration=True, indent=None, ignore_default_attributes=False, schema_location=None, no_namespace_schema_location=None, globalns=None)
key = 'pretty_print', value = True

    def __setattr__(self, key: str, value: Any):
        """Handle deprecated pretty print/indent behaviour."""
        if key == "pretty_print":
>           warnings.warn(
                "Setting `pretty_print` is deprecated, use `indent` instead",
                DeprecationWarning,
            )
E           DeprecationWarning: Setting `pretty_print` is deprecated, use `indent` instead

/opt/hostedtoolcache/Python/3.11.8/x64/lib/python3.11/site-packages/xsdata/formats/dataclass/serializers/config.py:47: DeprecationWarning
```

--- a/binder/requirements.txt
+++ b/binder/requirements.txt
@@ -244,9 +244,7 @@ traitlets==5.14.0
 triton==2.2.0
     # via torch
 typing-extensions==4.8.0
-    # via
-    #   torch
-    #   xsdata
+    # via torch
 urllib3==2.1.0
     # via requests
 wcwidth==0.2.12
@@ -255,8 +253,10 @@ wget==3.2
     # via zenodo-get
 wheel==0.42.0
     # via pip-tools
-xsdata==23.8
-    # via ismrmrd
+xsdata==22.12
+    # via
+    #   ismrmrd
+    #   mrpro (pyproject.toml)
 zenodo-get==1.5.1
     # via mrpro (pyproject.toml)
 

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -31,6 +31,7 @@ dependencies = [
     "numpy>=1.23,<2.0",
     "torch>=2.2,<3.0",
     "ismrmrd",
+    "xsdata>=22.2,<23",
     "einops",
     "pydicom",
     "pypulseq@git+https://github.com/imr-framework/pypulseq",

